sdk/cosmos/azure-cosmos/azure/cosmos/aio/user.py

```diff
@@ -148,16 +154,21 @@ def query_permissions(
         return result
 
     @distributed_trace_async
-    async def get_permission(self, permission, **kwargs):
-        # type: (Union[str, Dict[str, Any], Permission], Any) -> Permission
+    async def get_permission(
+            self,
+            permission: Union[str, Dict[str, Any], _Permission],
+            **kwargs: Any
+    ) -> _Permission:
```

Comment:
Unfortunately underscored typehints don't render as we'd like - is the reason we've imported this with an underscore to prevent it becoming duplicate API?
The best way to fix this - which we should do before GA regardless - is to make all these files private (i.e. `_user.py`) an only surface the parts we want through `__init__.py`. I know that will be inconsistent with the sync client... but we can't easily fix that, though I don't think it prevents us using a cleaner file structure for async.

In the short term, you can work around this with something like:
```python
from typing import TYPE_CHECKING:
if TYPE_CHECKING:
    from ..permission import Permission

async def get_permission(self, permission: "Permission") -> "Permission":
    ...
```

Reply:
Ah it looks like it's used in code as well as type hints - in which case I refer to my first suggestion :D

Reply:
makes sense, it seems like regardless another point was to make as many things as we can private so this aligns with that too